src/ncp/ncp_spinel.cpp

```diff
@@ -439,6 +497,183 @@ void NcpSpinel::HandleValueIs(spinel_prop_key_t aKey, const uint8_t *aBuffer, ui
     return;
 }
 
+#if OTBR_ENABLE_SRP_ADVERTISING_PROXY
+static std::string KeyNameFor(const otPlatDnssdKey &aKey)
+{
+    std::string name(aKey.mName);
+
+    if (aKey.mServiceType != nullptr)
+    {
+        name += ".";
+        name += aKey.mServiceType;
+    }
+    return name;
```

Comment:
We should be careful here, as the first label in a "service instance name" may itself contain a dot (`.`) character. So, if we construct a full name by appending the labels, it may be parsed incorrectly later.

Reply:
Added a check to only add the dot when the name doesn't contain one in the end.

Reply:
The `.` can be in the middle and part of the instance name label itself (not a separator). The dot `.`  is generally not used (or allowed) in DNS labels, but service instance labels allow this. For example, a service instance label can be "device.1234" as a single label (`.` being part of the label itself).

If we want to handle this, I can think of two options:

1.  Keep the first label separate and pass it as a separate "string" to all APIs (mDNS-related APIs).
2.  Add a new escaping mechanism so we can construct a full name as a single string, using an escape character to indicate when `.` is part of the label and not a separator.

I prefer option 1 (that's why we use this way to represent the names from the OT core).

I know this is rather tricky, so it may be fine to leave it for the future (accepting that the current code would not work with instance labels that include a `.`).

Reply:
SGTM! I added a comment in the code.